### PR TITLE
Feat/wgsfilter rules

### PR DESCRIPTION
**Pull request by ashwini06**
### This PR:

Adds new rules for filtering WGS TNscope and TNhaplotyper results.
Currently only applicable to tumor-only analysis. 

Problem:
- Not all vcf files are in the same format. Requires choice of different fields (MQ, DP, AF..) for the respective vcf file.

- [x] Finding the right fields for filtering from respective vcf file. 
- [ ] Finding the optimal threshold cutoffs.
- [x] bcftools intersect for finding the common variants called by TNscope and TNhaplotyper.

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by ashwini06** [comment]
Tested on WGS -tumor only case `civilsole`. Check reduction in the number of variants for files for `*.filtered.pass.vcf.gz`
a1

**Comment by hassanfa** [comment]
Tests are failing for `ModuleNotFoundError: No module named 'setuptools_rust'` We are not using Rust, probably one of the dependencies changed after merging master into develop. I'll check it out.

**Comment by ashwini06** [comment]
> Tests are failing for `ModuleNotFoundError: No module named 'setuptools_rust'` We are not using Rust, probably one of the dependencies changed after merging master into develop. I'll check it out.

@hassanfa Locally no issues with the installation
aa2

**Comment by hassanfa** [comment]
@ashwini06 try with -U option in pip. 

Apparently `gsutils` is using a new dependency, and that's causing a problem.

**Comment by ashwini06** [comment]
> @ashwini06 try with -U option in pip.
> 
> Apparently `gsutils` is using a new dependency, and that's causing a problem.

Indeed. So is it for balsamic report delivery or are we accessing any other resources from cloud?

**Comment by hassanfa** [comment]
`gsutils` is used on `reference.smk` to download various reference files.

**Comment by hassanfa** [comment]
@ashwini06 should we merge this and work on optimal threshold values later, or do you prefer to finish it all in this PR?

**Review by hassanfa — APPROVED** [review]
👌🏽

File: BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_only.rule [hunk 1]
```diff
@@ -35,3 +35,35 @@ tabix -p vcf -f {output.vcf_filtered};
...
+    pop_freq = [SENTIEON_CALLER.pop_freq.tag_value, SENTIEON_CALLER.pop_freq.filter_name],
+  threads: get_threads(cluster_config, 'ngs_filter_sentieon')
+  singularity: Path(singularity_image, config["bioinfo_tools"].get("bcftools") + ".sif").as_posix()
+  benchmark:
+    benchmark_dir + 'ngs_filter_' + "{var_type}.somatic.{case_name}.{var_caller}.vep.tsv"
+  shell:
+    """
+source activate {params.conda};
+bcftools view {input.vcf} \
+| bcftools filter --include 'FORMAT/AD[0:1]+AD[0:1] >= {params.DP[0]}' --soft-filter '{params.DP[1]}' --mode '+' \
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_only.rule:60]: This is not correct. According to http://samtools.github.io/bcftools/bcftools.html#expressions it should be:
```suggestion
| bcftools filter --include 'sum(FORMAT/AD[0:1]+FORMAT/AD[0:1]) >= {params.DP[0]}' --soft-filter '{params.DP[1]}' --mode '+' \
```

File: BALSAMIC/utils/constants.py [hunk 2]
```diff
@@ -188,6 +188,40 @@
     "description": "General purpose filters used for filtering VarDict",
 }
 
+# Configuration for SENTIEON settings:
+
+SENTIEON_VARCALL_SETTINGS = {
+    "AD": {
+        "tag_value": 10,
```
Comment (hassanfa) [inline BALSAMIC/utils/constants.py:195]:
```suggestion
        "tag_value": 2,
```

File: BALSAMIC/utils/constants.py [hunk 3]
```diff
@@ -188,6 +188,40 @@
...
+        "filter_name": "balsamic_low_tumor_dp",
+        "field": "FORMAT",
+    },
+    "AF_max": {
+        "tag_value": 1,
+        "filter_name": "balsamic_af_one",
+        "field": "FORMAT"
+    },
+    "AF_min": {
+        "tag_value": 0.01,
```
Comment (hassanfa) [inline BALSAMIC/utils/constants.py:210]:
```suggestion
        "tag_value": 0.05,
```

**Review by hassanfa — CHANGES_REQUESTED** [review]

File: BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_only.rule (hunk 1 above)
Comment (ashwini06) [reply BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_only.rule:60]: Fixed

File: BALSAMIC/utils/constants.py (hunk 2 above)
Comment (ashwini06) [reply BALSAMIC/utils/constants.py]: Fixed

File: BALSAMIC/utils/constants.py (hunk 3 above)
Comment (ashwini06) [reply BALSAMIC/utils/constants.py]: Fixed

**Comment by hassanfa** [comment]
@ashwini06 I realized I forgot to paste filter string here for posterity. So here it is:

```
 bcftools filter --regions-file callable SNV.somatic.civilsole.tnscope.pass.vcf.gz | bcftools filter --include 'SUM(FORMAT/AD[0:1] + FORMAT/AD[0:0])>10' --mode + --soft-filter filter552 | bcftools filter --include 'FORMAT/AD[0:1]>3 && (FORMAT/ALT_F1R2 > 0  && FORMAT/ALT_F2R1 > 0 &&  FORMAT/REF_F1R2 > 0 && FORMAT/REF_F2R1 > 0) ' --mode + --soft-filter filter112 | bcftools filter --include 'FORMAT/AF[0]>0.05' --mode + --soft-filter filter23123 | bcftools filter --include 'FORMAT/AF < 1' --soft-filter filter4 --mode + | bcftools filter --include 'INFO/GNOMADAF_popmax <= 0.001 || INFO/GNOMADAF_popmax == "."' --mode + --soft-filter filter3 | bcftools filter --include 'SUM(FORMAT/QSS)/SUM(FORMAT/AD) > 20' --mode + --soft-filter filter2 | bcftools view -f PASS | bcftools filter --include "INFO/SOR<2" --mode + --soft-filter blabla | bcftools +counts
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule [hunk 4]
```diff
@@ -2,34 +2,108 @@
...
-bcftools view {input.vcf} \
+grep -v '^@' {input.wgs_calling_file} > {input.wgs_calling_file}.bed
+bcftools filter --regions-file {input.wgs_calling_file}.bed {input.vcf} \
+| bcftools filter --include 'SUM(FORMAT/AD[0:0]+FORMAT/AD[0:1]) >= {params.DP[0]}' --soft-filter '{params.DP[1]}' --mode '+' \
+| bcftools filter --include 'FORMAT/AD[0:1] > {params.AD[0]}' --soft-filter '{params.AD[1]}' --mode '+' \
+| bcftools filter --include 'FORMAT/AF > {params.AF_min[0]}' --soft-filter '{params.AF_min[1]}' --mode '+' \
+| bcftools filter --include 'FORMAT/AF <  {params.AF_max[0]}' --soft-filter '{params.AF_max[1]}' --mode '+' \
+| bcftools filter --include 'INFO/GNOMADAF_popmax <= {params.pop_freq[0]} || INFO/GNOMADAF_popmax == \".\"' --soft-filter '{params.pop_freq[1]}' --mode '+' \
+| bcftools filter --include 'SUM(FORMAT/QSS)/SUM(FORMAT/AD) >= {params.qss[0]}' --soft-filter '{params.qss[1]}' --mode '+' \
+|  bcftools filter --include 'FORMAT/ALT_F1R2 > {params.cnt_reads[0]} && (FORMAT/ALT_F1R2 > 0  && FORMAT/ALT_F2R1 > {params.cnt_reads[0]} &&  FORMAT/REF_F1R2 > {params.cnt_reads[0]} && FORMAT/REF_F2R1 > {params.cnt_reads[0]})' --soft-filter '{params.cnt_reads[1]}' --mode '+' \
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:39]:
```suggestion
| bcftools filter --include 'FORMAT/ALT_F1R2 > {params.cnt_reads[0]} && (FORMAT/ALT_F1R2 > 0  && FORMAT/ALT_F2R1 > {params.cnt_reads[0]} &&  FORMAT/REF_F1R2 > {params.cnt_reads[0]} && FORMAT/REF_F2R1 > {params.cnt_reads[0]})' --soft-filter '{params.cnt_reads[1]}' --mode '+' \
```

Comment (hassanfa) [reply BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:39]: tiny extra space :-)

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule [hunk 5]
```diff
@@ -2,34 +2,108 @@
...
+  input:
+    tnscope = vep_dir + "{var_type}.somatic.{case_name}.tnscope.filtered.pass.vcf.gz",
+    tnhaplotyper = vep_dir + "{var_type}.somatic.{case_name}.tnhaplotyper.filtered.pass.vcf.gz"
+  output:
+    vcf_name = vep_dir + "{var_type}.somatic.{case_name}.tnscope.filtered.pass.clinical.vcf.gz"
+  params:
+    conda = config["bioinfo_tools"].get("bcftools"),
+    vcf_dir = vep_dir + "sentieon_callers_intersect"
+  threads: 
+    get_threads(cluster_config, 'ngs_filter_intersect')
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:97]: Is it possible to use threads in bcftools command?

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule [hunk 6]
```diff
@@ -2,34 +2,108 @@
...
+  singularity: 
+    Path(singularity_image, config["bioinfo_tools"].get("bcftools") + ".sif").as_posix()     
+  benchmark:
+    benchmark_dir + 'ngs_filter_' + "{var_type}.somatic.{case_name}.vep.tsv"
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:101]:
```suggestion
    benchmark_dir + 'ngs_filter_intersect_' + "{var_type}.somatic.{case_name}.tsv"
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule [hunk 7]
```diff
@@ -2,34 +2,108 @@
...
 bcftools view -f PASS -o {output.vcf_pass} -O z {output.vcf_filtered};
 tabix -p vcf -f {output.vcf_pass};
     """
+
+rule ngs_filter_intersect:
+  input:
+    tnscope = vep_dir + "{var_type}.somatic.{case_name}.tnscope.filtered.pass.vcf.gz",
+    tnhaplotyper = vep_dir + "{var_type}.somatic.{case_name}.tnhaplotyper.filtered.pass.vcf.gz"
+  output:
+    vcf_name = vep_dir + "{var_type}.somatic.{case_name}.tnscope.filtered.pass.clinical.vcf.gz"
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:92]:
```suggestion
    vcf_pass = vep_dir + "{var_type}.somatic.{case_name}.tnscope.filtered.pass.vcf.gz"
```

This is to make it unified with rest of final housekeeper tags.

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule [hunk 8]
```diff
@@ -2,34 +2,108 @@
...
+  params:
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:93]:
```suggestion
  params:
        housekeeper_id = {"id": config["analysis"]["case_id"], "tags": "clinical"},
```

Adds housekeeper tags

File: BALSAMIC/utils/constants.py [hunk 9]
```diff
@@ -212,10 +212,25 @@
...
+        "tag_value": 0.001,
         "filter_name": "balsamic_high_pop_freq",
         "field": "INFO"
     },
+    "qss": {
+	"tag_value": 20 ,
+	"filter_name": "balsamic_low_quality_scores",
+	"field": "FORMAT"
+    },
+    "cnt_reads": {
```
Comment (hassanfa) [inline BALSAMIC/utils/constants.py:224]:
```suggestion
    "strand_reads": {
```

To make it obvious that this is aimed for strand specific counts.

File: BALSAMIC/utils/constants.py [hunk 10]
```diff
@@ -212,10 +212,25 @@
...
+	"tag_value": 0 ,
+	"filter_name": "balsamic_low_read_counts",
```
Comment (hassanfa) [inline BALSAMIC/utils/constants.py:226]:
```suggestion
	"filter_name": "balsamic_low_strand_read_count",
```

File: BALSAMIC/utils/constants.py [hunk 11]
```diff
@@ -212,10 +212,25 @@
...
+	"field": "FORMAT"
+    },
+    "sor": {
+	"tag_value": 3,
+	"filter_name": "balsamic_sym_or",
```
Comment (hassanfa) [inline BALSAMIC/utils/constants.py:231]:
```suggestion
	"filter_name": "balsamic_high_strand_oddsratio",
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule [hunk 12]
```diff
@@ -2,34 +2,108 @@
...
+source activate {params.conda};
+grep -v '^@' {input.wgs_calling_file} > {input.wgs_calling_file}.bed
+bcftools filter --regions-file {input.wgs_calling_file}.bed {input.vcf} \
 | bcftools filter --include 'SUM(FORMAT/AD[0:0]+FORMAT/AD[0:1]) >= {params.DP[0]}' --soft-filter '{params.DP[1]}' --mode '+' \
 | bcftools filter --include 'FORMAT/AD[0:1] >= {params.AD[0]}' --soft-filter '{params.AD[1]}' --mode '+' \
 | bcftools filter --include 'FORMAT/AF >= {params.AF_min[0]}' --soft-filter '{params.AF_min[1]}' --mode '+' \
 | bcftools filter --include 'FORMAT/AF <  {params.AF_max[0]}' --soft-filter '{params.AF_max[1]}' --mode '+' \
 | bcftools filter --include 'INFO/GNOMADAF_popmax <= {params.pop_freq[0]} || INFO/GNOMADAF_popmax == \".\"' --soft-filter '{params.pop_freq[1]}' --mode '+' \
+| bcftools filter --include 'SUM(FORMAT/QSS)/SUM(FORMAT/AD) >= {params.qss[0]}' --soft-filter '{params.qss[1]}' --mode '+' \
+|  bcftools filter --include 'FORMAT/ALT_F1R2 > {params.cnt_reads[0]} && (FORMAT/ALT_F1R2 > 0  && FORMAT/ALT_F2R1 > {params.cnt_reads[0]} &&  FORMAT/REF_F1R2 > {params.cnt_reads[0]} && FORMAT/REF_F2R1 > {params.cnt_reads[0]})' --soft-filter '{params.cnt_reads[1]}' --mode '+' \
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:80]:
```suggestion
| bcftools filter --include 'FORMAT/ALT_F1R2 > {params.cnt_reads[0]} && (FORMAT/ALT_F1R2 > 0  && FORMAT/ALT_F2R1 > {params.cnt_reads[0]} &&  FORMAT/REF_F1R2 > {params.cnt_reads[0]} && FORMAT/REF_F2R1 > {params.cnt_reads[0]})' --soft-filter '{params.cnt_reads[1]}' --mode '+' \
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+| bcftools view -o {output.vcf_filtered} -O z;
+tabix -p vcf -f {output.vcf_filtered};
+bcftools view -f PASS -o {output.vcf_pass} -O z {output.vcf_filtered};
+tabix -p vcf -f {output.vcf_pass};
+    """
+
+rule ngs_filter_intersect:
+  input:
+    tnscope = vep_dir + "{var_type}.somatic.{case_name}.tnscope.filtered.pass.vcf.gz",
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:89]:
```suggestion
    tnscope = vep_dir + "{var_type}.somatic.{case_name}.tnscope.filtered.vcf.gz",
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+    tnhaplotyper = vep_dir + "{var_type}.somatic.{case_name}.tnhaplotyper.filtered.pass.vcf.gz"
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:90]:
```suggestion
    tnhaplotyper = vep_dir + "{var_type}.somatic.{case_name}.tnhaplotyper.filtered.vcf.gz"
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+# coding: utf-8
+# NGS filters for various scenarios
+
+rule ngs_filter_tnscope:
+  input:
+    vcf = vep_dir + "{var_type}.somatic.{case_name}.tnscope.pass.vcf.gz",
+    wgs_calling_file = config["reference"]["wgs_calling_interval"]
+  output:
+    vcf_filtered = vep_dir + "{var_type}.somatic.{case_name}.tnscope.filtered.vcf.gz",
+    vcf_pass = vep_dir + "{var_type}.somatic.{case_name}.tnscope.filtered.pass.vcf.gz",
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:11]:
```suggestion
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+| bcftools filter --include 'FORMAT/AF > {params.AF_min[0]}' --soft-filter '{params.AF_min[1]}' --mode '+' \
+| bcftools filter --include 'FORMAT/AF <  {params.AF_max[0]}' --soft-filter '{params.AF_max[1]}' --mode '+' \
+| bcftools filter --include 'INFO/GNOMADAF_popmax <= {params.pop_freq[0]} || INFO/GNOMADAF_popmax == \".\"' --soft-filter '{params.pop_freq[1]}' --mode '+' \
+| bcftools filter --include 'SUM(FORMAT/QSS)/SUM(FORMAT/AD) >= {params.qss[0]}' --soft-filter '{params.qss[1]}' --mode '+' \
+|  bcftools filter --include 'FORMAT/ALT_F1R2 > {params.cnt_reads[0]} && (FORMAT/ALT_F1R2 > 0  && FORMAT/ALT_F2R1 > {params.cnt_reads[0]} &&  FORMAT/REF_F1R2 > {params.cnt_reads[0]} && FORMAT/REF_F2R1 > {params.cnt_reads[0]})' --soft-filter '{params.cnt_reads[1]}' --mode '+' \
+| bcftools filter --include "INFO/SOR < {params.sor[0]}" --soft-filter '{params.sor[1]}' --mode '+' \
+| bcftools view -o {output.vcf_filtered} -O z;
+tabix -p vcf -f {output.vcf_filtered};
+bcftools view -f PASS -o {output.vcf_pass} -O z {output.vcf_filtered};
+tabix -p vcf -f {output.vcf_pass};
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:44]:
```suggestion
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+| bcftools filter --include 'FORMAT/AD[0:1] >= {params.AD[0]}' --soft-filter '{params.AD[1]}' --mode '+' \
+| bcftools filter --include 'FORMAT/AF >= {params.AF_min[0]}' --soft-filter '{params.AF_min[1]}' --mode '+' \
+| bcftools filter --include 'FORMAT/AF <  {params.AF_max[0]}' --soft-filter '{params.AF_max[1]}' --mode '+' \
+| bcftools filter --include 'INFO/GNOMADAF_popmax <= {params.pop_freq[0]} || INFO/GNOMADAF_popmax == \".\"' --soft-filter '{params.pop_freq[1]}' --mode '+' \
+| bcftools filter --include 'SUM(FORMAT/QSS)/SUM(FORMAT/AD) >= {params.qss[0]}' --soft-filter '{params.qss[1]}' --mode '+' \
+|  bcftools filter --include 'FORMAT/ALT_F1R2 > {params.cnt_reads[0]} && (FORMAT/ALT_F1R2 > 0  && FORMAT/ALT_F2R1 > {params.cnt_reads[0]} &&  FORMAT/REF_F1R2 > {params.cnt_reads[0]} && FORMAT/REF_F2R1 > {params.cnt_reads[0]})' --soft-filter '{params.cnt_reads[1]}' --mode '+' \
+| bcftools view -o {output.vcf_filtered} -O z;
+tabix -p vcf -f {output.vcf_filtered};
+bcftools view -f PASS -o {output.vcf_pass} -O z {output.vcf_filtered};
+tabix -p vcf -f {output.vcf_pass};
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:84]:
```suggestion
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+    """
+
+rule ngs_filter_tnhaplotyper:
+  input:
+    vcf = vep_dir + "{var_type}.somatic.{case_name}.tnhaplotyper.pass.vcf.gz",
+    wgs_calling_file = config["reference"]["wgs_calling_interval"]
+  output:
+    vcf_filtered = vep_dir + "{var_type}.somatic.{case_name}.tnhaplotyper.filtered.vcf.gz",
+    vcf_pass = vep_dir + "{var_type}.somatic.{case_name}.tnhaplotyper.filtered.pass.vcf.gz",
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:53]:
```suggestion
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+  threads: 
+    get_threads(cluster_config, 'ngs_filter_intersect')
+  singularity: 
+    Path(singularity_image, config["bioinfo_tools"].get("bcftools") + ".sif").as_posix()     
+  benchmark:
+    benchmark_dir + 'ngs_filter_' + "{var_type}.somatic.{case_name}.vep.tsv"
+  shell:
+    """
+source activate {params.conda};
+bcftools isec {input.tnscope} {input.tnhaplotyper} -p {params.vcf_dir} -O z ;
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:105]:
```suggestion
bcftools isec {input.tnscope} {input.tnhaplotyper} -p {params.vcf_dir} -O z -f PASS;
```

To only take `PASS` variants

**Review by hassanfa — CHANGES_REQUESTED** [review]
👌🏽  Very close to the end.

Added some suggestions to `cnt_reads` and its filter name. Also to `sor` filter name.

I also removed `vcf_pass` from ngs filters tnscope and tnhaplotyper, then by passing `-f PASS` to isec, it'll only take passed variants.

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule (hunk 7 above)
Comment (ashwini06) [reply BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:92]: @hassanfa : The name will be same as {input.tnscope}.

Comment (hassanfa) [reply BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:92]: See my comments on that rule as well. Remove pass from file name:
`tnscope = vep_dir + "{var_type}.somatic.{case_name}.tnscope.filtered.vcf.gz",`

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule (hunk 8 above)
Comment (ashwini06) [reply BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:93]: @hassanfa how this params is used in bcftools command line?

Comment (hassanfa) [reply BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:93]: This will be picked up by delivery script, and used there.

**Review by hassanfa — APPROVED** [review]
👍🏽

**Pull request merged**